tt_metal/impl/dispatch/kernels/eth_tunneler.cpp

```diff
@@ -151,6 +152,10 @@ void kernel_main() {
             all_outputs_finished &= output_finished;
         }
 
+        tt_l1_ptr launch_msg_t * const launch_msg = GET_MAILBOX_ADDRESS_DEV(launch);
```

Comment:
Under what conditions will `exit_erisc_kernel` be set?
What will happen to kernels running on other cores in this chip as well as the neighboring chip that this ethernet tunneler is talking to?

Normally tunneler sends remote finished notification to each of its 2 inputs:
` input_queues[i].send_remote_finished_notification();`

When exit is triggered by exit_erisc_kernel, the code is not sending finish notification to its inputs (cores sending data to tunneler).

It's probably fine if the ethernet tunneler kernel on the ethernet neighbor chip will also be signaled to exit.
Want to confirm the high level flow/operation of `exit_erisc_kernel`.

Reply:
Logic for signaling `exit_erisc_kernel` is in `Device::reset_cores()`. It should match the logic for when we init the dispatch/tunneler cores: for each device, init dispatch cores required by it (including dispatch cores on pcie device servicing this remote device) + init tunneler cores on this device. Thinking is that it matches the init sequence so we don't run into any cases where we early exit a dispatch core we just set up.

`exit_erisc_kernel` is only signaled if we detect the kernel is still running (run msg == GO and exit_erisc_kernel == 0).